src/components/button/index.jsx

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable react/button-has-type */
+/*
+	Disabled ESLint rule as the type attribute is using oneOf propType with a default
+	value - which ESLint is unable to parse as acceptable when in the use case below
+	is valid as there is a default and the use of oneOf PropType validation
+*/
+import { forwardRef } from "react";
+import PropTypes from "prop-types";
+
+const COMPONENT_CLASS_NAME = "c-button";
+
+const Button = forwardRef((props, ref) => {
+	const {
+		accessibilityLabel,
+		children,
+		className,
+		disabled,
+		fullWidth,
+		href,
+		iconLeft,
+		iconRight,
+		size,
+		type,
+		variant,
+		...rest
+	} = props;
+
+	const defaultProps = {
+		...rest,
+		ref,
+		"aria-label": accessibilityLabel,
+		className: [
+			COMPONENT_CLASS_NAME,
+			href && `${COMPONENT_CLASS_NAME}--link`,
+			fullWidth && `${COMPONENT_CLASS_NAME}--full-width`,
+			size && `${COMPONENT_CLASS_NAME}--${size}`,
+			variant && `${COMPONENT_CLASS_NAME}--${variant}`,
+			className,
+		]
+			.filter((i) => i)
+			.join("  "),
```

Comment:
```suggestion
			.join(" "),
```

we want one space not two between classes, right?

Reply:
Doesn't wildly make a difference but just noticed it